### ISSUE TITLE
Support plugins in node_modules subdirectories

### ISSUE THREAD

**Issue by neoziro** [AtomLinter/linter-eslint#242]
I have that configuration:

```
node_modules/
  .bin/eslint -> ../linter/node_modules/.bin/eslint
  linter/
    node_modules/
      .bin/eslint -> ../eslint/bin/eslint.js
      eslint/
      eslint-plugin-react/
      eslint-plugin-mocha/
```

If I run in command line `eslint .`, it works great.

But in my editor I can see "[Linter-ESLint] plugin eslint-plugin-react not found".

It would be nice if you can check if the plugin is in the eslint directory and not in the directory of the project.

It would be very useful for the purpose of using a global linter for several projects.

**Comment by SpainTrain** [comment]
Where is that directory located? (project dir, global npm dir?)  And how did it come to have that structure?  (Installing https://www.npmjs.com/package/linter would not appear to create that directory structure)  What is the output of `which eslint`?

**Comment by neoziro** [comment]
In fact it's a private module to globalize linter and .eslintrc around all of our modules.

**Comment by neoziro** [comment]
It should be `./node_modules/@organization/linter/node_modules/.bin/eslint`.

**Comment by steelbrain** [comment]
@neoziro I am afraid you are mistaken my friend, `linter` isn't an [`npm`](https://npmjs.org) package, It's an [apm](https://atom.io/packages) package. We do however have a `atom-linter` npm package that works as a helper module and has a few commonly used methods for linter providers. Also if you are using the global eslint then you **do not** need to install `eslint-plugin-react` in your `~/.atom/packages/linter-eslint` directory. You can just install it globally and it'll be detected by the package or in a place where it's accessible by your eslint installation.

**Comment by neoziro** [comment]
@steelbrain no, I will rewrite the exact example:

```js
node_modules/
  .bin/eslint -> ../@doctolib/linter/node_modules/.bin/eslint
  @doctolib/
    linter/
      node_modules/
        .bin/eslint -> ../eslint/bin/eslint.js
        eslint/
        eslint-plugin-react/
        eslint-plugin-mocha/
```

I don't want to use global linter because it still specific to each project. But I would like to be able to use an eslint not directly in local deps but in sub dependency of another module. I hope that my example is clearer for you.

**Comment by steelbrain** [comment]
I still don't know what `linter` in your example above means. If you are referring to the atom package linter it's installed in `~/.atom/packages/linter` and **no** module requires it. They do however **export linter provider services** for that linter to use.

**Comment by neoziro** [comment]
It refers to this project (https://github.com/doctolib/linter), actually I use peerDependencies because it doesn't work right with atom. I would be able to use dependencies (not peerDependencies) and expose eslint binary.

**Comment by neoziro** [comment]
The goal is to share a global eslintrc config around our projects to avoid duplicating eslintrc config in all projects, this is a basic version and after I will improve but having it working in atom is a requirement for me.

**Comment by SpainTrain** [comment]
> The goal is to share a global eslintrc config

If the goal is to share *config* then why do you want to share the eslint *executable*?  Also, can you please run `which eslint` in the CLI so I can properly help you?

I think I understand what you are trying to accomplish, and recommend either:

* Install eslint and plugins as dev deps of your projects (because, well, they *are* dev deps of your project :hankey: )

* Create another install script like `copy-config` that installs the desired versions of eslint and plugins in the project

**Comment by steelbrain** [comment]
Just a side note of how I manage my config, I have a package called `eslint-config-steelbrain` installed globally, It has all the config. Then whenever I work on in a project I do
```js
{
  extends: "steelbrain"
}
```

**Comment by neoziro** [comment]
@steelbrain OK interesting, I don't want to use global packages but the extends can be a good option.

**Comment by neoziro** [comment]
@SpainTrain the problem is not finding eslint, the problem is finding plugins. You look for plugins in the project and not in subdepdencies. My case is probably an edge case, I just submitted it because it works when I use "eslint ." and not in editor which is weird. I can't give you the output of "which eslint" because as I said I use peerDependencies to solve my issue, so I don't have the previous version describe in my example.

**Comment by steelbrain** [comment]
Here's my proposed "workaround", at the moment we are requiring eslint from the atom `renderer` process which is a bad thing anyway, we could spawn the eslint executable like what happens in a shell, a use it's output from there

**Comment by neoziro** [comment]
@steelbrain yes I think just using eslint in PATH is simpler.

**Comment by steelbrain** [comment]
Fixed by v5.0.0
